Userland/Libraries/LibWeb/CSS/CSSFontFaceRule.cpp

```diff
@@ -31,8 +31,77 @@ CSSStyleDeclaration* CSSFontFaceRule::style()
 // https://www.w3.org/TR/cssom/#ref-for-cssfontfacerule
 String CSSFontFaceRule::serialized() const
 {
-    // FIXME: Implement this!
-    return "@font-face { /* FIXME: Implement CSSFontFaceRule::serialized()! */ }";
+    StringBuilder builder;
+    // The result of concatenating the following:
+
+    // 1. The string "@font-face {", followed by a single SPACE (U+0020).
+    builder.append("@font-face { "sv);
+
+    // 2. The string "font-family:", followed by a single SPACE (U+0020).
+    builder.append("font-family: "sv);
+
+    // 3. The result of performing serialize a string on the rule’s font family name.
+    builder.append(m_font_face.font_family());
+
+    // 4. The string ";", i.e., SEMICOLON (U+003B).
+    builder.append(";"sv);
+
+    // 5. If the rule’s associated source list is not empty, follow these substeps:
+    if (!m_font_face.sources().is_empty()) {
+        // 1. A single SPACE (U+0020), followed by the string "src:", followed by a single SPACE (U+0020).
+        builder.append("src: "sv);
+
+        // 2. The result of invoking serialize a comma-separated list on performing serialize a URL or serialize a LOCAL for each source on the source list.
```

Comment:
Would be cool if `serialize a comma-separated list` was its own function, to be reused. You don't have to do that now though, just thinking out loud.

Reply:
Hmm, I'm not sure how I would make a general purpose function from this as I'm not sure how to make a function that accepts a Vector of any type. Would such a function be able to be used in this context? Perhaps if it had a callback that would pass the current item?

Reply:
You could make it a template function with the Vector content as a template argument, and the callback as another one. Though the more I think about it, the less worth it it seems. :thonk:

Something like this should work:

```c++
template<typename T, typename SerializeItem>
void serialize_a_comma_separated_list(StringBuilder& builder, Vector<T> const& items, SerializeItem serialize_item) {
	// Iterate the items and call serialize_item() on them.
}
```

Templates have to completely go in the .h file.

Reply:
I'll do that, thanks for your help!